llvm/lib/Support/DXILABI.cpp

```diff
@@ -25,10 +25,14 @@ static const EnumEntry<dxil::ResourceClass> ResourceClassNames[] = {
     {"Sampler", llvm::dxil::ResourceClass::Sampler},
 };
 
-ArrayRef<EnumEntry<llvm::dxil::ResourceClass>> dxil::getResourceClasses() {
+ArrayRef<EnumEntry<llvm::dxil::ResourceClass>> getResourceClasses() {
   return ArrayRef(ResourceClassNames);
 }
 
 StringRef dxil::getResourceClassName(dxil::ResourceClass RC) {
   return enumToStringRef(RC, getResourceClasses());
 }

```

Comment:
Using ScopedPrinter for this seems like overkill if we aren't using the `EnumEntry` thing for anything else. Would it be better to just implement `dxil::getResourceClassName` as a simple switch statement?